File: app/lib/api/entity/TransportationEntity.dart
import 'package:app/api/entity/CityEntity.dart';
import 'package:app/api/entity/PropertiesEntity.dart';
import 'package:app/api/entity/UserEntity.dart';
import 'package:app/api/entity/enums/TransportationCategory.dart';
import 'package:json_annotation/json_annotation.dart';

part 'TransportationEntity.g.dart';

@JsonSerializable()
class TransportationEntity {
  int id;
  UserEntity creator;
  CityEntity city;
  AddressEntity addressTo;
  AddressEntity addressFrom;
  bool active;
  bool outCity;
  double price;
  String description;
  TransportationCategory category;
  DateTime date;
  DateTime createDate;
  @override
  bool operator ==(Object other) =>
      other is TransportationEntity && other.id == id;
  TransportationEntity(
      {required this.id,
      required this.creator,
      required this.city,
      required this.addressTo,
      required this.addressFrom,
      required this.active,
      required this.outCity,
      required this.category,
      required this.price,
      required this.description,
      required this.date,
      required this.createDate});

  factory TransportationEntity.fromJson(Map<String, dynamic> json) =>
      _$TransportationEntityFromJson(json);
  Map<String, dynamic> toJson() => _$TransportationEntityToJson(this);
}

File: server/GNext/src/main/java/com/thedeveloper/gnext/entity/TransportationEntity.java
package com.thedeveloper.gnext.entity;

import com.thedeveloper.gnext.enums.TransportationCategory;
import jakarta.persistence.*;
import lombok.Data;

import java.util.Date;

@Entity
@Data
@Table(name = "transportings")
public class TransportationEntity {
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    Long id;
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "creator", nullable = true, referencedColumnName = "phone")
    UserEntity creator;
    @ManyToOne
    @JoinColumn(name = "city", nullable = true,referencedColumnName = "name")
    CityEntity city;
    @ManyToOne
    @JoinColumn(name = "addressTo", nullable = true,referencedColumnName = "id")
    AddressEntity addressTo;
    @ManyToOne
    @JoinColumn(name = "addressFrom", nullable = true,referencedColumnName = "id")
    AddressEntity addressFrom;
    boolean active = true;
    boolean outCity;
    @Enumerated(EnumType.STRING)
    TransportationCategory category = TransportationCategory.ev;
    double price;
    String description;
    Date date;
    Date createDate;
}
